// ng6-countdown/bundles/countdown.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('moment'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ng6-countdown', ['exports', '@angular/core', 'moment', '@angular/common'], factory) :
    (factory((global.countdown = {}),global.ng.core,global.moment_,global.ng.common));
}(this, (function (exports,core,moment_,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NumberTransformPipe = /** @class */ (function () {
        function NumberTransformPipe() {
        }
        /**
         * @param {?} value
         * @return {?}
         */
        NumberTransformPipe.prototype.transform = /**
         * @param {?} value
         * @return {?}
         */
            function (value) {
                if (value) {
                    if (value < 10) {
                        return "0" + value;
                    }
                    return "" + value;
                }
                else {
                    return '00';
                }
            };
        NumberTransformPipe.decorators = [
            { type: core.Pipe, args: [{
                        name: 'appNumberTransform'
                    },] }
        ];
        return NumberTransformPipe;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var CountDownComponent = /** @class */ (function () {
        function CountDownComponent() {
            this.title = 'The time remaining is:';
            this.daysText = 'Days';
            this.hoursText = 'Hours';
            this.minutesText = 'Minutes';
            this.secondsText = 'Seconds';
            this.isCustomTemplate = false;
            this.cardBackgroundColor = 'seagreen';
            this.cardTextColor = 'white';
            this.daysChanged = new core.EventEmitter();
            this.hoursChanged = new core.EventEmitter();
            this.minutesChanged = new core.EventEmitter();
            this.secondsChanged = new core.EventEmitter();
            this.moment = moment_;
        }
        /**
         * @return {?}
         */
        CountDownComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this.finishDate = this.moment(this.date);
                setInterval(function () {
                    /** @type {?} */
                    var today = _this.moment();
                    _this.daysRemaining = _this.finishDate.diff(today, 'days') % 365;
                    _this.hoursRemaining = _this.finishDate.diff(today, 'hours') % 24;
                    _this.minutesRemaining = _this.finishDate.diff(today, 'minutes') % 60;
                    _this.secondsRemaining = _this.finishDate.diff(today, 'seconds') % 60;
                    _this.daysChanged.emit(_this.daysRemaining);
                    _this.hoursChanged.emit(_this.hoursRemaining);
                    _this.minutesChanged.emit(_this.minutesRemaining);
                    _this.secondsChanged.emit(_this.secondsRemaining);
                    _this.isFinishDate = _this.finishDate.isSameOrBefore(today);
                }, 1000);
            };
        CountDownComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'lib-countdown',
                        template: "<div *ngIf=\"!isFinishDate && !isCustomTemplate\" class=\"hide-on-small-only\">\n    <div class=\"row container\">\n      <div class=\"col s12 center-align\">\n        <h2 class=\"text-bold\"> {{title}} </h2>\n      </div>\n    </div>\n    <div class=\"row container\">\n      <div class=\"col s3\">\n        <div class=\"card\">\n          <div class=\"card-image waves-effect waves-block waves-light\">\n            <p class=\"number-style center-align\"> {{daysRemaining | appNumberTransform}} </p>\n          </div>\n          <div class=\"card-content\" [style.background]=\"cardBackgroundColor\">\n            <span class=\"card-title activator text-darken-4 center-align\" [style.color]=\"cardTextColor\">{{daysText}}</span>\n          </div>\n        </div>\n      </div>\n      <div class=\"col s3\">\n        <div class=\"card\">\n          <div class=\"card-image waves-effect waves-block waves-light\">\n            <p class=\"number-style center-align\"> {{hoursRemaining | appNumberTransform}} </p>\n          </div>\n          <div class=\"card-content\" [style.background]=\"cardBackgroundColor\">\n            <span class=\"card-title activator text-darken-4 center-align\" [style.color]=\"cardTextColor\">{{hoursText}}</span>\n          </div>\n        </div>\n      </div>\n      <div class=\"col s3\">\n        <div class=\"card\">\n          <div class=\"card-image waves-effect waves-block waves-light\">\n            <p class=\"number-style center-align\"> {{minutesRemaining | appNumberTransform}} </p>\n          </div>\n          <div class=\"card-content\" [style.background]=\"cardBackgroundColor\">\n            <span class=\"card-title activator text-darken-4 center-align\" [style.color]=\"cardTextColor\">{{minutesText}}</span>\n          </div>\n        </div>\n      </div>\n      <div class=\"col s3\">\n        <div class=\"card\">\n          <div class=\"card-image waves-effect waves-block waves-light\">\n            <p class=\"number-style center-align\"> {{secondsRemaining | appNumberTransform}} </p>\n          </div>\n          <div class=\"card-content\" [style.background]=\"cardBackgroundColor\">\n            <span class=\"card-title activator text-darken-4 center-align\" [style.color]=\"cardTextColor\">{{secondsText}}</span>\n          </div>\n        </div>\n      </div>\n    </div>\n  </div>\n  <div *ngIf=\"!isFinishDate && !isCustomTemplate\" class=\"row container show-on-small hide-on-med-and-up\">\n    <div class=\"row container\">\n      <div class=\"col s12 center-align\">\n        <h5 class=\"text-bold count-title-mobile\"> {{title}} </h5>\n      </div>\n    </div>\n    <div class=\"col s12\">\n      <div class=\"card\">\n        <div class=\"card-image waves-effect waves-block waves-light\">\n          <p class=\"number-style-mobile center-align\"> \n            <span class=\"text-bold\">{{daysText}}: </span> {{daysRemaining | appNumberTransform}} | \n            <span class=\"text-bold\"> {{hoursText}}: </span> {{hoursRemaining | appNumberTransform}} |\n            <span class=\"text-bold\"> {{minutesText}}:</span> {{minutesRemaining | appNumberTransform}} |\n            <span class=\"text-bold\"> {{secondsText}}:</span> {{secondsRemaining | appNumberTransform}} </p>\n        </div>\n      </div>\n    </div>\n  </div>\n  <ng-content *ngIf=\"isCustomTemplate\"></ng-content>",
                        styles: [".number-style{font-size:60px;font-weight:700}.number-style-mobile{font-size:20px}@media screen and (max-width:415px){.number-style-mobile{font-size:13px!important}}@media screen and (max-width:380px){.number-style-mobile{font-size:12px!important}}@media screen and (max-width:330px){.number-style-mobile{font-size:10px!important}}.text-bold{font-weight:700}@media screen and (max-width:415px){.card-title{font-size:20px!important}}@media screen and (max-width:380px){.card-title{font-size:15px!important}}@media screen and (max-width:330px){.card-title{font-size:10px!important}}@media screen and (max-width:360px){.count-title-mobile{font-size:20px!important}}@media screen and (max-width:320px){.count-title-mobile{font-size:18px!important}}"]
                    }] }
        ];
        CountDownComponent.propDecorators = {
            date: [{ type: core.Input }],
            title: [{ type: core.Input }],
            daysText: [{ type: core.Input }],
            hoursText: [{ type: core.Input }],
            minutesText: [{ type: core.Input }],
            secondsText: [{ type: core.Input }],
            isCustomTemplate: [{ type: core.Input }],
            cardBackgroundColor: [{ type: core.Input }],
            cardTextColor: [{ type: core.Input }],
            daysChanged: [{ type: core.Output }],
            hoursChanged: [{ type: core.Output }],
            minutesChanged: [{ type: core.Output }],
            secondsChanged: [{ type: core.Output }]
        };
        return CountDownComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var CountDownModule = /** @class */ (function () {
        function CountDownModule() {
        }
        CountDownModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [
                            CountDownComponent,
                            NumberTransformPipe,
                        ],
                        imports: [
                            common.CommonModule
                        ],
                        exports: [
                            CountDownComponent
                        ]
                    },] }
        ];
        return CountDownModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.CountDownModule = CountDownModule;
    exports.ɵa = CountDownComponent;
    exports.ɵb = NumberTransformPipe;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
